docs/architecture/adr-002-configv2.md

```diff
@@ -0,0 +1,109 @@
+# ADR 2: Configuration v2
+
+## Overview
+
+This ADR is very high level proposal for the overall goals and high level aspirations for configuration v2. As such the puspose is not to go overly deep into where/how different fields are defined.
+
+The examples provided in this ADR are currently not exhaustive. So do NOT pay too much attention to each field, they're meant to be more examples of the _types_ of configuration for each object.
+
+## Context
+
+The configration of k0s has organically growed from very small to something quite complex today. While that is natural we're starting to see growing number of issues stemmin from the complexity.
+
+- Lack of separation between per node configuration and cluster configuration
+- Lack of feedback in dynamic config reconciliation status
+- Lack of feedback in helm and stack applier status (should these be in the clusterconfig to begin with?)
+- No separation between used defined configuration and the configuration that is used internally
+- Lack of versioning
+- Dynamic config has several problems when it comes to reconciliation of certain fields
+- Some fields have side effects that aren't very predictable.
+- The k0s' configuration for calico doesn't really match the calico configuration
+- Config doesn't allow to disable components, it requires modifying the cmdline, specially with dynamic config, it would make sense to allow this from the configuration.
```

Comment:
```suggestion
- Config doesn't allow users to to disable components - doing so requires modifying the cmdline. With dynamic config, it would make sense to allow this from the configuration.
```